File: 프로그래머스/2/42578. 의상/의상.py
from itertools import combinations

def solution(clothes):
    
    kinds = {} 
    
    cnt = 1
    
    for c in clothes:
        if c[1] not in kinds.keys():
            kinds[c[1]] = [[]]
        kinds[c[1]].append(c[0])
        
    print(kinds)
    
    
    # 각 종류의 경우의 수 다 곱하기
#     for k in kinds.keys():
#         cnt *= len(kinds[k])
#     # print(cnt)
    
        
#     # 각 분야에서 하나씩만 입을 때
#     if len(kinds.keys()) > 1: 
#         for k in kinds.keys():
#             cnt += len(kinds[k])
        
    # print(cnt)
    
    
    for k in kinds.keys():
        cnt *= len(kinds[k])
    
    return cnt - 1

        
    

File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*;

class Solution {
    
    static Map<String, ArrayList<String>>  map = new HashMap<>();
    static int n = 0;
    static int answer = 1;

    
    // 조합 변수
    // static ArrayList<String> answer = new ArrayList<>();
    // static ArrayList<String> result = new ArrayList<>();
    
    
    
    public int solution(String[][] clothes) {
        n = clothes.length;
        
        for(int i = 0; i<n; i++){
            map.putIfAbsent(clothes[i][1], new ArrayList<>());
            map.get(clothes[i][1]).add(clothes[i][0]);
        }
        
        
        
        for(String key: map.keySet()){
            ArrayList<String> values = map.get(key);
            answer *= values.size()+1;
            System.out.println(values);
        }


        
        return answer-1;
    }
    
    
    
//     static public backtrack(int dep, ArrayList<String> list, int pre){
        
//         if 
        
        
//     }
    
    
    
    
    
    
    
    
}